docker/synapse.Dockerfile

```diff
@@ -20,4 +20,8 @@ RUN /venv/bin/pip install -q --no-cache-dir lxml psycopg2 coverage codecov
 # and test
 RUN /venv/bin/pip uninstall -q --no-cache-dir -y matrix-synapse
 
+# Pre-install test dependencies installed by `scripts/synapse_sytest.sh`.
+RUN /venv/bin/pip install -q --no-cache-dir \
+        lxml psycopg2 coverage codecov tap.py coverage_enable_subprocess
+
```

Comment:
The relevant part of `scripts/synapse_sytest.sh` is
https://github.com/matrix-org/sytest/blob/20521395ee42104edb6ed76aa33a9ce9329c08ed/scripts/synapse_sytest.sh#L132-L151